src/module/rules/rule-element/grant-item/rule-element.ts

```diff
@@ -193,6 +206,52 @@ class GrantItemRuleElement extends RuleElementPF2e {
         return noAction;
     }
 
+    /** Is the item-alteration data structurally sound? Currently only overrides are supported. */
+    #isValidItemAlteration(data: {}): data is ItemAlterationData[] {
+        return (
+            Array.isArray(data) &&
+            data.every(
+                (d: unknown) =>
+                    d instanceof Object &&
+                    "mode" in d &&
+                    d.mode === "override" &&
+                    "property" in d &&
+                    d.property === "badge-value" &&
+                    "value" in d &&
+                    (["string", "number"].includes(typeof d.value) || d.value === null)
+            )
+        );
+    }
+
+    /** Is the item alteration valid for the granted item type? */
+    #itemCanBeAltered(grantedSource: ItemSourcePF2e, value: unknown): boolean | null {
+        const sourceId = grantedSource.flags.core?.sourceId ? ` (${grantedSource.flags.core.sourceId})` : "";
+        if (grantedSource.type !== "condition" && grantedSource.type !== "effect") {
+            this.failValidation(`unable to alter "${grantedSource.name}"${sourceId}: must be condition or effect`);
+            return false;
+        }
+
+        const hasBadge =
+            grantedSource.type === "condition"
+                ? typeof grantedSource.system.value.value === "number"
+                : grantedSource.type === "effect"
+                ? grantedSource.system.badge?.type === "counter"
```

Comment:
Even if we get more badge types, changing the value should still be ok (badge type only declares that it can be incremented/decrement) but we can cross that bridge when we get there.